qa/tasks/mgr/test_module_selftest.py

```diff
@@ -236,10 +236,10 @@ def test_module_commands(self):
         # Calling a command on a disabled module should return the proper
         # error code.
         self.mgr_cluster.mon_manager.raw_cluster_cmd(
-            "mgr", "module", "disable", "status")
+            "mgr", "module", "disable", "hello")
         with self.assertRaises(CommandFailedError) as exc_raised:
             self.mgr_cluster.mon_manager.raw_cluster_cmd(
-                "osd", "status")
```

Comment:
http://pulpito.ceph.com/kchai-2018-08-21_15:56:38-rados-wip-kefu-testing-2018-08-21-1839-distro-basic-mira/2932436/

Reply:
http://tracker.ceph.com/issues/26994

Reply:
The hello module isn't installed, so despite _definitely_ being disabled, it can't announce its commands.

Reply:
something like this will probably work

```diff
diff --git a/qa/tasks/mgr/test_module_selftest.py b/qa/tasks/mgr/test_module_selftest.py
index fa609f7348..dd3c3123a4 100644
--- a/qa/tasks/mgr/test_module_selftest.py
+++ b/qa/tasks/mgr/test_module_selftest.py
@@ -231,18 +231,18 @@ class TestModuleSelftest(MgrTestCase):
         disabled/failed/recently-enabled modules.
         """
 
-        self._load_module("selftest")
-
         # Calling a command on a disabled module should return the proper
         # error code.
         self.mgr_cluster.mon_manager.raw_cluster_cmd(
-            "mgr", "module", "disable", "status")
+            "mgr", "module", "disable", "selftest")
         with self.assertRaises(CommandFailedError) as exc_raised:
             self.mgr_cluster.mon_manager.raw_cluster_cmd(
-                "osd", "status")
+                "mgr", "self-test", "run")
 
         self.assertEqual(exc_raised.exception.exitstatus, errno.EOPNOTSUPP)
 
+        self._load_module("selftest")
+
         # Calling a command that really doesn't exist should give me EINVAL.
         with self.assertRaises(CommandFailedError) as exc_raised:
             self.mgr_cluster.mon_manager.raw_cluster_cmd(
```

Reply:
thanks @noahdesu , i am posting #23681 on behalf of you.